src/simulator_core/adapter/transforms/mappers.py

```diff
@@ -126,7 +128,17 @@ def to_entity(self, model: EsdlObject) -> NetworkController:
         This method first converts all assets into a list of assets.
         Next to this a list of Junctions is created. This is then used
         to create the NetworkController object.
-        :param EsdlObject entity: EsdlObject object to be converted to NetworkController object
+        :param EsdlObject model: EsdlObject object to be converted to NetworkController object
         :return: NetworkController, which is the converted EsdlObject object.
         """
-        return NetworkController()
+        consumers = []
+        for esdl_asset in model.get_all_assets_of_type("consumer"):
+            consumers.append(ControllerConsumer(esdl_asset.esdl_asset.name,
+                                                esdl_asset.esdl_asset.id))
+            consumers[-1].add_profile(esdl_asset.get_profile())
```

Comment:
what happens when the esdl_asset doesnt have a profile?

Reply:
Dan wordt het een puinhoop, want get_profile geeft dan niks terug.
Wat willen we moet elke consumer een profile hebben?
Voor nu zou ik zeggen ja, dus dan gooi ik wel een exception

Reply:
Prima gefixt met raise error in **esdl_asset_object.py**